src/t_stream.c

```diff
@@ -1441,9 +1456,29 @@ void xrangeGenericCommand(client *c, int rev) {
     long long count = -1;
     robj *startarg = rev ? c->argv[3] : c->argv[2];
     robj *endarg = rev ? c->argv[2] : c->argv[3];
-
-    if (streamParseIDOrReply(c,startarg,&startid,0) == C_ERR) return;
-    if (streamParseIDOrReply(c,endarg,&endid,UINT64_MAX) == C_ERR) return;
+    int startex = 0, endex = 0, invalid_id = 0;
+    char *startp = startarg->ptr, *endp = endarg->ptr;
+    size_t startlen = sdslen(startarg->ptr), endlen = sdslen(endarg->ptr);
+
+    /* Check for open/close (inclusive/exclusive) range prefixes. */
+    if (startlen > 1 && startp[0] == '(' && startp[1] != '-') {
+        startex = 1;
+    }
+    if (endlen > 1 && endp[0] == '(' && endp[1] != '-') {
+        endex = 1;
+    }
+    
+    /* Parse start and end IDs. */
+    startarg = createStringObject(startp+startex,startlen-startex);
+    endarg = createStringObject(endp+endex,endlen-endex);
+    invalid_id = (streamParseIDOrReply(c,startarg,&startid,0) == C_ERR) ||
+                 (streamParseIDOrReply(c,endarg,&endid,UINT64_MAX) == C_ERR);
+    decrRefCount(startarg);
+    decrRefCount(endarg);
+
+    if (invalid_id) return;
```

Comment:
this is a bit fragile. maybe at least trim the empty line between the decrement and the return.
i.e. the `OrReply` above is coupled with the `return`